src/instance-group-manager.js

```diff
@@ -153,6 +153,64 @@ class InstanceGroupManager extends common.ServiceObject {
      */
     this.name = name;
   }
+  /**
+   * Flags the specified instances to be removed from the managed instance group.
+   * @see [InstanceGroupManagers: abandonInstances API Documentation]{@link https://cloud.google.com/compute/docs/reference/v1/instanceGroupManagers/abandonInstances}
+   * @param {VM|VM[]} vms - VM instances to abandon from
+   *     this instance group manager.
+   * @param {function} callback - The callback function.
+   * @param {?error} callback.err - An error returned while making this request.
+   * @param {Operation} callback.operation - An operation object
+   *     that can be used to check the status of the request.
+   * @param {object} callback.apiResponse - The full API response.
+   *
+   * @example
+   * const Compute = require('@google-cloud/compute');
+   * const compute = new Compute();
+   * const zone = compute.zone('us-central1-a');
+   * const instanceGroupManager = zone.instanceGroupManager('web-servers');
+   *
+   * const vms = [
+   *   gce.zone('us-central1-a').vm('http-server'),
+   *   gce.zone('us-central1-a').vm('https-server')
+   * ];
+   *
+   * instanceGroupManager.abandonInstances(vms, function(err, operation, apiResponse) {
+   *   // `operation` is an Operation object that can be used to check the status
+   *   // of the request.
+   * });
+   *
+   * //-
+   * // If the callback is omitted, we'll return a Promise.
+   * //-
+   * instanceGroupManager.abandonInstances(vms).then(function(data) {
+   *   const operation = data[0];
+   *   const apiResponse = data[1];
+   * });
+   */
+  abandonInstances(vms, callback) {
+    const self = this;
+    this.request(
```

Comment:
Thanks for this 👍 I know that you copied the code from the `deleteInstances` method, where these suggestions aren't used, but since we're putting in new code, could you make these little changes?

```suggestion
this.request(
    {
      method: 'POST',
      uri: '/abandonInstances',
      json: {
        instances: arrify(vms).map(vm => vm.url), // format change
      },
    },
    (err, resp) => { // save the scope, so no `self` required!
      if (err) {
        callback(err, null, resp);
        return;
      }
      const operation = this.zone.operation(resp.name); // later, self!
      operation.metadata = resp;
      callback(null, operation, resp);
    }
  );
}
```

Reply:
Fixed in next patch.